integrations/ultralytics/deepsparse/deepsparse_utils.py

```diff
@@ -102,3 +119,65 @@ def postprocess_nms(outputs: torch.Tensor) -> List[numpy.ndarray]:
     # run nms in PyTorch, only post-process first output
     nms_outputs = non_max_suppression(outputs)
     return [output.cpu().numpy() for output in nms_outputs]
+
+
+def modify_yolo_onnx_input_shape(
+    model_path: str, image_shape: Tuple[int]
+) -> Tuple[str, Optional[NamedTemporaryFile]]:
+    """
+
+    :param model_path: file path to YOLOv3 ONNX model or SparseZoo stub of the model
+        to be loaded
+    :param image_shape: 2-tuple of the image shape to resize this yolo model to
+    :return: filepath to an onnx model reshaped to the given input shape and tempfile
+        object that the modified model is written to. if the given model has the given
+        input shape, then the path to the original model will be returned with no
+        tempfile. tempfile returned so caller can control when the tempfile is destroyed
+    """
+    original_model_path = model_path
+    if model_path.startswith("zoo:"):
+        # load SparseZoo Model from stub
+        model = Zoo.load_model_from_stub(model_path)
+        model_path = model.onnx_file.downloaded_path()
+
+    model = onnx.load(model_path)
+    model_input = model.graph.input[0]
+
+    initial_x = _get_onnx_tensor_idx_shape(model_input, 2)
+    initial_y = _get_onnx_tensor_idx_shape(model_input, 3)
+
+    if not (isinstance(initial_x, int) and isinstance(initial_y, int)):
+        return model_path, None  # model graph does not have static integer input shape
+
+    if (initial_x, initial_y) == tuple(image_shape):
+        return model_path, None  # no shape modification needed
+
+    scale_x = initial_x / image_shape[0]
+    scale_y = initial_y / image_shape[1]
+    _set_onnx_tensor_idx_shape(model_input, 2, image_shape[0])
+    _set_onnx_tensor_idx_shape(model_input, 3, image_shape[1])
+
+    for model_output in model.graph.output:
+        output_x = _get_onnx_tensor_idx_shape(model_output, 2)
+        output_y = _get_onnx_tensor_idx_shape(model_output, 3)
+        _set_onnx_tensor_idx_shape(model_output, 2, int(output_x / scale_x))
+        _set_onnx_tensor_idx_shape(model_output, 3, int(output_y / scale_y))
+
+    tmp_file = NamedTemporaryFile()  # file will be deleted after program exit
+    onnx.save(model, tmp_file.name)
+
+    print(
+        f"Overwriting original model shape {(initial_x, initial_y)} to {image_shape}\n"
+        f"Original model path: {original_model_path}, new temporary model saved to "
+        f"{tmp_file.name}"
+    )
+
+    return tmp_file.name, tmp_file
+
+
+def _get_onnx_tensor_idx_shape(tensor: onnx.TensorProto, idx: int) -> int:
```

Comment:
I think it would be good to move this get and set to generic utils in the sparseml.onnx package

Reply:
good point. updated